### DISCUSSION TITLE
Extracting camera params from EXIF and XMP.

### DISCUSSION THREAD

**Discussion by alfieroddan** [bigcat88/pillow_heif#225] [Q&A] [answered]
Hi,

Thank you very much for the work on this, really great work!

I am interested in using this package for computer vision projects. The HEIC format is able to save the camera parameters, see the "more info section" below:

<img width="394" alt="Screenshot 2024-03-27 at 12 37 39" src="https://github.com/bigcat88/pillow_heif/assets/51797647/37ef5b85-30f1-44f1-a03f-45337918cfd0">

This is really useful information for computer vision, just wondering if anyone can help me extract using this package.

My attempt so far is below:

```python
from PIL import Image, ExifTags
from pillow_heif import register_heif_opener

register_heif_opener()

# read heic image
img = Image.open("/Users/alfie/Library/CloudStorage/OneDrive-ImperialColleg
eLondon/WP2/tracking/leica-da-vinci-markertest-26-03-2024/IMG_3538.HEIC")

exif = { ExifTags.TAGS[k]: v for k, v in img.getexif().items() if k in Exif
Tags.TAGS }

print(exif)

print(img.getxmp())
```

Output:

```bash
(py12) ~/Documents/tmp$ python heic_test.py 
{'ResolutionUnit': 2, 'ExifOffset': 216, 'Make': 'Apple', 'Model': 'iPhone 12 mini', 'Software': '17.2.1', 'Orientation': 1, 'DateTime': '2024:03:26 11:40:46', 'XResolution': 72.0, 'YResolution': 72.0, 'HostComputer': 'iPhone 12 mini'}
{}
```

Any help would be appreciated!

**Comment by bigcat88** [comment, marked as answer]
```shell

examples % python3 pillow_dump_info.py
```

Look at example's code:

```python3
        exif = image.getexif()
        if exif:
            exif = exif._get_merged_dict()  # noqa
        for k, v in exif.items():
            if isinstance(v, bytes) and len(v) > 8:
                print(f"{TAGS[k]} : size {len(v)} bytes")
            else:
                print(TAGS[k], ":", v)
```

But as I know new Pillow versions has something better to get `Exif` info.
You can take a look at examples here: https://github.com/python-pillow/Pillow/issues/5863

**Reply by alfieroddan** [reply]
Worked perfectly! Thanks
